src/intel_rdt.c

```diff
@@ -158,12 +159,24 @@ static void rdt_submit(const struct pqos_mon_data *group) {
   if (events & PQOS_PERF_EVENT_LLC_REF) {
     uint64_t value;
 
-    int ret = pqos_mon_get_value(group, PQOS_PERF_EVENT_LLC_REF, &value, NULL);
+    int ret = pqos_mon_get_value(group, PQOS_PERF_EVENT_LLC_REF, NULL, &value);
     if (ret == PQOS_RETVAL_OK)
-      rdt_submit_gauge(desc, "bytes", "llc_ref", value);
+      rdt_submit_derive(desc, "bytes", "llc_ref", value);
   }
 #endif
 
+  if (events & PQOS_PERF_EVENT_LLC_MISS) {
+#if PQOS_VERSION >= 40400
+    uint64_t value;
+
+    int ret = pqos_mon_get_value(group, PQOS_PERF_EVENT_LLC_MISS, NULL, &value);
+    if (ret == PQOS_RETVAL_OK)
+      rdt_submit_derive(desc, "bytes", "llc_miss", value);
+#else
+    rdt_submit_derive(desc, "bytes", "llc_miss", values->llc_misses_delta);
```

Comment:
I've never used collectd with derived/delta metrics, only gauges & counters. Do collectd write plugins delta values to totals?

Otherwise there could be a problem...

Write plugins do not know whether same client queried the value earlier, so they could report the latest available value on every query, meaning that one cannot add them up in case one got same metric multiple times. Or if plugin reports given metric only once, and multiple clients are querying for them, then some of the delta values are missed, and client cannot construct total/avg for given (longer) time interval.

@octo Any advice on using derive delta vs. total counter values for metrics?

Reply:
llc_misses is a counter and llc_misses_delta is difference from previous pull. Counter value should be reported as derive